### ISSUE TITLE
Missing messages from self in chats

### ISSUE THREAD

**Issue by TheGuardianWolf** [UnigramDev/Unigram#1335]
May be related to #1334 .

During the loading process for large chats, your more recent messages can sometimes disappear completely from chat history. A reload of the chat is required.

This is somewhat hard to test as it requires posting into a large chat to reproduce...

I'll continue chatting over the day to see if there's a more reliable way to trigger this.

**Comment by TheGuardianWolf** [comment]
Found a possible way to reproduce:

1. Reply to a message
2. Click a link to another chat that would trigger #1334
3. Click back into the group chat

In the below image, I clicked the GitHub bot link which I confirmed to trigger #1334, and then I clicked back into the Unigram insiders chat, where my reply to the issue is missing. It is not missing on another Telegram client.

![Unigram insiders missing chat](https://i.imgur.com/Yp5BMb1.png)

![Telegram X chat](https://i.imgur.com/En931h8.png)
